lib/oelite/task.py

```diff
@@ -183,24 +183,17 @@ def colon_split(s):
         self._meta = meta
         return meta
 
-
-    def run(self):
+    def prepare_context(self):
         meta = self.meta()
-        function = meta.get_function(self.name)
-
+        self.function = meta.get_function(self.name)
         self.do_cleandirs()
-        cwd = self.do_dirs() or meta.get("B")
-
-        # Setup stdin, stdout and stderr redirection
-        stdin = open("/dev/null", "r")
-        self.logfn = "%s/%s.%s.log"%(function.tmpdir, self.name, meta.get("DATETIME"))
-        self.logsymlink = "%s/%s.log"%(function.tmpdir, self.name)
+        self.cwd = self.do_dirs() or meta.get("B")
+        self.stdin = open("/dev/null", "r")
+        self.logfn = "%s/%s.%s.log"%(self.function.tmpdir, self.name, meta.get("DATETIME"))
+        self.logsymlink = "%s/%s.log"%(self.function.tmpdir, self.name)
         oelite.util.makedirs(os.path.dirname(self.logfn))
         try:
-            if self.debug:
-                logfile = os.popen("tee %s"%self.logfn, "w")
```

Comment:
meh, I really do not like to loose that feature, it is highly annoying that you have to dig into logfiles if you are running a specific build to just quickly see what is happening e.g. for configure

However, I'm pretty sure the way the output forwarding was done with tee, introduced a massive slowdown to the build, so that was obviously not nice anyway

It is not necessary to force it into serial mode here to show output. The Exherbo package manager already implements this nicely, but muxing between the output from the tasks been run, showing each of them for $user-configurable-amount-of-time. This works just fine in practice